src/uu/shuf/src/shuf.rs

```diff
@@ -59,13 +65,17 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
             }
         }
     } else {
-        Mode::Default(
-            matches
-                .get_one::<String>(options::FILE)
-                .map(|s| s.as_str())
-                .unwrap_or("-")
-                .to_string(),
-        )
+        let mut operands = matches
+            .get_many::<String>(options::FILE_OR_ARGS)
+            .unwrap_or_default();
+        let file = operands.next().cloned().unwrap_or("-".into());
+        if let Some(second_file) = operands.next() {
+            return Err(UUsageError::new(
+                1,
+                format!("unexpected argument '{second_file}' found"),
+            ));
+        };
```

Comment:
Can you please add a test for this error case?

The other changes look good :)